src/codegen.rs

```diff
@@ -54,6 +54,9 @@ mod tests;
 /// A wrapper around the LLVM context to allow passing it without exposing the inkwell dependencies
 pub struct CodegenContext(Context);
 
+unsafe impl Send for CodegenContext {}
+unsafe impl Sync for CodegenContext {}
```

Comment:
Correct me if I'm wrong, but these two impls are not needed afaik? It seems we always generate the `CodegenContext` locally within threads, without sharing them accross multiple ones?